grails-app/controllers/smartR/plugin/SmartRController.groovy

```diff
@@ -0,0 +1,81 @@
+package smartR.plugin
+
+import grails.converters.JSON
+import org.codehaus.groovy.grails.web.json.JSONObject
+import org.codehaus.groovy.grails.web.json.JSONArray
+import groovy.json.JsonBuilder
+import org.apache.commons.io.FilenameUtils
+
+
+class SmartRController {
+
+    def smartRService
+
+    /**
+    *   Renders the default view
+    */
+    def index = {
+        def dir = smartRService.getScriptsFolder()
+        def scriptList = new File(dir).list().findAll { it != 'Wrapper.R' && it != 'Sample.R' }
```

Comment:
I don't like that the controller is listing files. Surely it can get this from a service.